docs/walk-through/loops.md

```diff
@@ -2,13 +2,41 @@
 
 When writing workflows, it is often very useful to be able to iterate over a set of inputs, as this is how argo-workflows can perform loops.
 
-There are two basic ways of running a template multiple times.
+There are three basic ways of running a template multiple times.
 
+- `withSequence` iterates over a sequence of numbers.
 - `withItems` takes a list of things to work on. Either
     - plain, single values, which are then usable in your template as '{{item}}'
     - a JSON object where each element in the object can be addressed by it's key as '{{item.key}}'
 - `withParam` takes a JSON array of items, and iterates over it - again the items can be objects like with `withItems`. This is very powerful, as you can generate the JSON in another step in your workflow, so creating a dynamic workflow.
 
+## `withSequence` example
+
+This runs a template multiple times using `withSequence`.
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: loop-sequence-
+spec:
+  entrypoint: loop-sequence-example
+
+  templates:
+  - name: loop-sequence-example
+    steps:
+    - - name: hello-world-x5
+        template: hello-world
+        withSequence:
+          count: "5"
+
+  - name: run-pod
+    container:
+      image: argoproj/argosay:v2
+      command: ["sh", -c]
+      args: ["echo hello world!"]
+```
+
 ## `withItems` basic example
 
 This example is the simplest. We are taking a basic list of items and iterating over it with `withItems`. It is limited to one varying field for each of the workflow templates instantiated.
```

Comment:
```suggestion
This iterates over a list of items with `withItems`, substituting a string for each instantiated template.
```

this line was what I mentioned in my [previous comment](https://github.com/argoproj/argo-workflows/pull/12577#discussion_r1495365093). it's no longer the "simplest example" and there are other changes that should be made here according to the style guide